### PR TITLE
add basic Docker-Coq-Action opam-based CI for Coq master

### PR DESCRIPTION
**Pull request by palmskog**
Includes a small badge in README.md that shows CI status.

### CODE REVIEW

**Comment by damien-pous** [comment]
Just merged, thanks a lot!, now waiting to see if it actually passes the CI check...

Do you know of a way to further setup the CI 
1. to also check my pushes on branch 8.17 (w.r.t. Coq 8.17)
2. in a way where the CI-related files on the repo are shared between my master and 8.17 branches

??

File: README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Coinduction
 
+[![Docker CI][docker-action-shield]][docker-action-link]
+[docker-action-shield]: https://github.com/damien-pous/coinduction/workflows/Docker%20CI/badge.svg?branch=master
+[docker-action-link]: https://github.com/damien-pous/coinduction/actions?query=workflow:"Docker%20CI"
```
Comment (palmskog) [inline README.md:5]: Sorry, there needs to be a newline between the first and second line here for the badge to work. That is, it should be as follows:
```
[![Docker CI][docker-action-shield]][docker-action-link]

[docker-action-shield]: https://github.com/damien-pous/coinduction/workflows/Docker%20CI/badge.svg?branch=master
[docker-action-link]: https://github.com/damien-pous/coinduction/actions?query=workflow:"Docker%20CI"
```
Maybe easiest if you push this directly.

Comment (damien-pous) [reply README.md:5]: sure, done.

**Comment by palmskog** [comment]
1. Sure, the Docker-Coq-Action CI can be configured to run differently on different branches, and also differently for PRs targeting different branches. Here is an example for ATBR: https://github.com/coq-community/atbr/blob/v8.17/.github/workflows/docker-action.yml#L5-L11
2. Our current best way of sharing CI configuration between projects and branches is through our [templates](https://github.com/coq-community/templates). A file `meta.yml` contains some data about which branch it resides in and compatible Coq versions, and generates (for example) CI boilerplate through the [mustache](https://mustache.github.io/) system. See an example in ATBR: https://github.com/coq-community/atbr/blob/v8.17/meta.yml#L10

**Comment by palmskog** [comment]
I think it's probably hard to do what you want to do completely friction-free.

But the files `meta.yml`, `docker-action.yml`, etc., would practically never be modified by the Coq devs (they generally only change `.ml`/`.mli`/`.v`). So you could rely on Git to relatively easily merge changes from `master` to go on top of the boilerplate differences between the branches. Some aspects of GitHub Actions can be configured outside repo data I believe, but our standard setup is based on modifying `.yml` files in `.github/workflows`.

**Pull request merged**